src/debug.cpp

```diff
@@ -893,9 +894,9 @@ static cata::optional<uintptr_t> debug_compute_load_offset(
             return cata::nullopt;
         }
 
-        char buf[1024];
-        while( fgets( buf, sizeof( buf ), nm ) ) {
-            std::string line( buf );
+        std::array<char, 1024> buf;
+        while( fgets( buf.data(), sizeof( buf ), nm ) ) {
```

Comment:
You should probably use `buf.size()` as there is no guarantee that `sizeof(std::array<T, N>)` actually equals N.

Reply:
I think it actually is guaranteed, but `.size()` is clearer, so I'll change it.